### PR TITLE
Add Java analyzer for Annalyns Infiltration

### LINKED ISSUE #98: annalyns-infiltration: implement analyzer
Implement an analyzer for the concept exercise `annalyns-infiltration`.
Details on the feedback that should be provided by the analyzer can be found in the exercise's `.meta/design.md` file, see https://github.com/exercism/java/issues/2669.

### PR DESCRIPTION
**Pull request by Gyoo**
Fixes #98 

I also added the analyzer comments to the website-copy repository : https://github.com/exercism/website-copy/pull/2323

### CODE REVIEW

**Comment by manumafe98** [comment]
Hey Gyoo, thanks for submitting a PR! I would suggest to change this to draft, and wait until this [major reformat](https://github.com/exercism/java-analyzer/pull/124) of the analyzer is applied

**Comment by Gyoo** [comment]
Thank you for your feedback. I will monitor the other issue for now :)

**Comment by sanderploegsma** [comment]
Hi @Gyoo, we have finished refactoring the test setup, so feel free to update your branch with the changes on `main` and finishing this PR 👍

File: src/test/resources/analyzer/AnalyzerIntegrationTest.annalynsinfiltration.Optimal.approved.txt
Comment (sanderploegsma) [inline src/test/resources/analyzer/AnalyzerIntegrationTest.annalynsinfiltration.Optimal.approved.txt:1]: According to the exercise's [.meta/design.md](https://github.com/exercism/java/blob/main/exercises/concept/annalyns-infiltration/.meta/design.md) an optimal solution should receive a `celebratory` comment, similar to the other concept exercises.

**Review by sanderploegsma — CHANGES_REQUESTED** [review]
Please also add a couple of smoke tests for this exercise to the `tests/` directory. We aim to have at least two: one with an exemplar solution and one or more with a solution that is designed to receive some feedback from the analyzer.

Have a look at the smoke tests for the other exercises to see what is expected.

**Comment by Gyoo** [comment]
@sanderploegsma I made the changes you requested but I'm wondering if I'm missing something regarding smoke tests : I have one exemplar solution and 4 other tests to trigger the different analyzer cases. Do I need something more ?

**Comment by sanderploegsma** [comment]
The main difference between the JUnit tests in `src/test/java` and the smoke tests in `tests/` is that the former is used to verify the correctness of the analyzer and achieve test coverage, but it does not run the full analyzer end-to-end like it would run on Exercism. This is why we have the smoke tests: they work by running the analyzer on a solution as a Docker container, similar to how the website runs it.

**Comment by Gyoo** [comment]
Thank you for your feedback. I'll have a second look at this then!

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM gradle:8.5-jdk21 as build
+FROM gradle:8.6-jdk21 as build
```
Comment (sanderploegsma) [inline Dockerfile:1]: Any particular reason for updating this? I'd rather do this in a separate PR as we try to keep the Gradle versions in sync between the Docker file and the Gradle wrapper.

File: tests/annalyns-infiltration/exemplar-solution/expected_analysis.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{
+  "comments": [
+    {
+      "comment": "java.general.exemplar",
+      "params": {
+        "exerciseName": "Annalyn\u0027s Infiltration"
```
Comment (sanderploegsma) [inline tests/annalyns-infiltration/exemplar-solution/expected_analysis.json:6]: Any idea why the apostrophe is printed as a unicode character here? I'm not sure whether this will render correctly when presented on the website. Maybe @ErikSchierboom can shine some light on whether this is supported.

File: Dockerfile (hunk 1 above)
Comment (Gyoo) [reply Dockerfile:1]: Actually  I can remove that. I had issues running the docker test runner for smoke tests because of this image and when I checked on dockerhub it looked like 8.5-jdk21 wasn't available anymore but that's actually not the case.

**Review by sanderploegsma — APPROVED** [review]
Other than my one open question this looks good to go, thanks!

Note for maintainers: make sure the website-copy PR is merged _first_

File: tests/annalyns-infiltration/exemplar-solution/expected_analysis.json (hunk 2 above)
Comment (ErikSchierboom) [reply tests/annalyns-infiltration/exemplar-solution/expected_analysis.json:6]: I think it will be fine.

**Review by ErikSchierboom — APPROVED** [review]

**Comment by sanderploegsma** [comment]
Thanks for your contribution @Gyoo! 🎉
